internal/tiltfile/tiltfile_state.go

```diff
@@ -1372,6 +1384,34 @@ func (s *tiltfileState) setTeam(thread *starlark.Thread, fn *starlark.Builtin, a
 	return starlark.None, nil
 }
 
+func (s *tiltfileState) dockerPruneSettings(thread *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var disable bool
+	var intervalHrs, numBuilds, maxAgeMins int
+	if err := s.unpackArgs(fn.Name(), args, kwargs,
+		"disable?", &disable,
+		"max_age_mins?", &maxAgeMins,
+		"num_builds?", &numBuilds,
+		"interval_hrs?", &intervalHrs); err != nil {
+		return nil, err
+	}
+
+	if disable && (intervalHrs != 0 || numBuilds != 0 || maxAgeMins != 0) {
+		return nil, fmt.Errorf("can't disable Docker Prune (`disabled=True`) and pass additional settings")
```

Comment:
Why not?

Reply:
> ah, sorry, this PR has gotten way too big to review, i thought the plan was to break it up?

idr us discussing that but sure, can do

Reply:
cuz it doesn't make sense? if we're not ever running docker prune, why would we allow you to configure settings for it? (i don't feel strongly about this tho.)

Reply:
I could imagine it being annoying if you wanted to do something like this:

```python
pruner_disabled = read_config()
docker_prune_settings(disable=pruner_disabled, interval_hrs=2)
```

With how the API is now you'd have to instead rwrite
```python
pruner_disabled = read_config()
if pruner_disabled:
  docker_prune_settings(disable=pruner_disabled, interval_hrs=2)
```

Not a big deal, but that's my logic.

Reply:
mmm fair.